<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ShaderManagerWidget</class>
 <widget class="QWidget" name="ShaderManagerWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>272</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ShaderManagerWidget</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Shaders options</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Lighting mode</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="fragmentShaderChoice"/>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QStackedWidget" name="shaderConfiguration">
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="page">
         <layout class="QFormLayout" name="formLayout_2">
          <property name="fieldGrowthPolicy">
           <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
          </property>
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
         </layout>
        </widget>
        <widget class="QWidget" name="page_3"/>
        <widget class="QWidget" name="page_2">
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="title">
             <string>Ponctual light</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <item>
              <widget class="VectorChoiceWidget" name="directionPonctualLight" native="true"/>
             </item>
             <item>
              <widget class="QWidget" name="widget_2" native="true">
               <layout class="QFormLayout" name="formLayout_3">
                <property name="leftMargin">
                 <number>0</number>
                </property>
                <property name="topMargin">
                 <number>0</number>
                </property>
                <property name="rightMargin">
                 <number>0</number>
                </property>
                <property name="bottomMargin">
                 <number>0</number>
                </property>
                <item row="0" column="0">
                 <widget class="QLabel" name="label_5">
                  <property name="text">
                   <string>Ambient</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="ColorChoiceWidget" name="ambientPonctualLight" native="true"/>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="label_6">
                  <property name="text">
                   <string>Diffuse</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="ColorChoiceWidget" name="diffusePonctualLight" native="true"/>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="label_7">
                  <property name="text">
                   <string>Specular</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="ColorChoiceWidget" name="specularPonctualLight" native="true"/>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ColorChoiceWidget</class>
   <extends>QWidget</extends>
   <header>colorchoicewidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>VectorChoiceWidget</class>
   <extends>QWidget</extends>
   <header>vectorchoicewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>fragmentShaderChoice</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>shaderConfiguration</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>235</x>
     <y>32</y>
    </hint>
    <hint type="destinationlabel">
     <x>199</x>
     <y>75</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
